### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by mrsamsharma**
## What does this PR do
In this PR me and my coding partner check the error about the Testing and make the changes.
## Description of the task to be completed
- [x] Write unit test for all components  
- [x] Test the operate.js and calculate.js file
- [x] Used snapshot 

## How should it be tested
Clone the repository
```
git clone https://github.com/mrsamsharma/Math-Magicians.git
```
Navigate to the folder
```
cd Math-Magicians
```
git checkout Testing
```
#### Thanks for reviewing my PR 😊

### CODE REVIEW

**Review by ha-manel — COMMENTED** [review]
# STATUS: INVALID 🔴 
Hi @mrsamsharma 
I'm sure you put a lot of effort into this project but this PR is invalid because this is a pair programming activity but you've worked on it alone😓your partner needs to have some commits in this project too. If you have permission to work alone kindly include a screenshot in the pull request 👍  

Kindly fix the issue and submit the PR for review again 👍 
Best of luck!

------
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/test/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import calculate from '../logic/calculate';
```
Comment (FrederickMih) [inline src/test/calculate.test.js:1]: - [ ] I think there're still some scenarios that you haven't yet tested. Based on the requirement of this project, you are expected to test all functionalities and scenarios of this module.
- Kindly also considered the case where the buttonName is equal to i.e `buttonName === '+/-', '=', '.'(dot) `

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: @Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Screenshot from 2022-10-20 09-38-51




## Highlights :green_circle: 

- Tests are passing :heavy_check_mark: 
- App still works smoothly :heavy_check_mark: 
- Descriptive PR :+1: 

## Required Changes ♻️

- [ ] For better mastering of the concept(testing), let's the other team member kindly fix the changes requested below :arrow_double_down: 
- [ ] Also, let's kindly add snapshots tests for all the components(inside the page directory)

_Check the comments under the review._

## Optional suggestions

- [Optional] The other contributor is missing in the Author section of the README.md file. Since it is a collaborative project, let's include all team members.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by gemmen29 — APPROVED** [review]
Hi @mrsamsharma, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**